tests/lib/semanticInfo.js

```diff
@@ -0,0 +1,182 @@
+    expect(boundName.name).toBe('x');
+
+    const tsBoundName = parseResult.services.esTreeNodeToTSNodeMap.get(
+      boundName
+    );
+    expect(tsBoundName).toBeDefined();
+
+    checkNumberArrayType(checker, tsBoundName);
+
+    expect(parseResult.services.tsNodeToESTreeNodeMap.get(tsBoundName)).toBe(
+      boundName
+    );
+  });
+
+  test('imported-file tests', () => {
+    const fileName = path.resolve(FIXTURES_DIR, 'import-file.src.ts');
+    const parseResult = testUtils.parseCode(
+      shelljs.cat(fileName),
+      createOptions(fileName)
+    );
+
+    // get type checker
+    expect(parseResult).toHaveProperty('services.program.getTypeChecker');
+    const checker = parseResult.services.program.getTypeChecker();
+
+    // get array node (ast shape validated by snapshot)
+    // node is defined in other file than the parsed one
+    const arrayBoundName = parseResult.ast.body[1].expression.callee.object;
+    expect(arrayBoundName.name).toBe('arr');
+
+    expect(parseResult).toHaveProperty('services.esTreeNodeToTSNodeMap');
+    const tsArrayBoundName = parseResult.services.esTreeNodeToTSNodeMap.get(
+      arrayBoundName
+    );
+    expect(tsArrayBoundName).toBeDefined();
+    checkNumberArrayType(checker, tsArrayBoundName);
+
+    expect(
+      parseResult.services.tsNodeToESTreeNodeMap.get(tsArrayBoundName)
+    ).toBe(arrayBoundName);
+  });
+
+  test('non-existent project file', () => {
+    const fileName = path.resolve(FIXTURES_DIR, 'isolated-file.src.ts');
+    const badConfig = createOptions(fileName);
+    badConfig.project = './tsconfigs.json';
+    expect(() =>
+      testUtils.parseCode(shelljs.cat(fileName), badConfig)
+    ).toThrowErrorMatchingSnapshot();
+  });
+
+  test('fail to read project file', () => {
+    const fileName = path.resolve(FIXTURES_DIR, 'isolated-file.src.ts');
+    const badConfig = createOptions(fileName);
+    badConfig.project = '.';
+    expect(() =>
+      testUtils.parseCode(shelljs.cat(fileName), badConfig)
+    ).toThrowErrorMatchingSnapshot();
+  });
+
+  test('malformed project file', () => {
+    const fileName = path.resolve(FIXTURES_DIR, 'isolated-file.src.ts');
+    const badConfig = createOptions(fileName);
+    badConfig.project = './badTSConfig/tsconfig.json';
+    expect(() =>
+      testUtils.parseCode(shelljs.cat(fileName), badConfig)
+    ).toThrowErrorMatchingSnapshot();
+  });
+});
+
+/**
+ * Verifies that the type of a TS node is number[] as expected
+ * @param {ts.TypeChecker} checker
+ * @param {ts.Node} tsNode
+ */
+function checkNumberArrayType(checker, tsNode) {
+  const nodeType = /** @type {ts.ObjectType & ts.TypeReference} */ (checker.getTypeAtLocation(
+    tsNode
+  ));
+  expect(nodeType.flags).toBe(ts.TypeFlags.Object);
+  expect(nodeType.objectFlags).toBe(ts.ObjectFlags.Reference);
+  expect(nodeType.typeArguments).toHaveLength(1);
+  expect(nodeType.typeArguments[0].flags).toBe(ts.TypeFlags.Number);
```

Comment:
Currently, this doesn't *technically* test that you have an array, though it's close. It just tests you have a

```maff
∀ Foo: * → *. Foo<number>
```

Reply:
Discussed offline; unless I misunderstood our conclusion, we decided that since the global array type is not part of the public API, this is the best it makes sense to do.